### ISSUE TITLE
Rule proposal: Prefer lookahead and lookbehind

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1113]
Avoid capture unneeded groups.

## Fail

```js
'I love unicorn! I hate unicorn?'.replace(/(?<before>love )unicorn(?<after>!)/, '$<before>🦄$<after>');
```

```js
'I love unicorn! I hate unicorn?'.replace(/(love )unicorn(!)/, '$1🦄$2');
```

## Pass

```js
'I love unicorn! I hate unicorn?'.replace(/(?<=love )unicorn(?=!)/, '🦄');
```

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by fisker** [comment]
The rule already implemented as [`regexp/prefer-lookaround`](https://ota-meshi.github.io/eslint-plugin-regexp/rules/prefer-lookaround.html).
